contribs/gnodev/pkg/browser/client_dev.go

```diff
@@ -0,0 +1,120 @@
+package browser
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/gnolang/gno/contribs/gnodev/pkg/emitter"
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/gorilla/websocket"
+)
+
+const MaxElapsedTime = time.Second * 20
+
+var ErrHandlerNotSet = errors.New("handler not set")
+
+type DevClient struct {
+	Logger  *slog.Logger
+	Handler func(typ events.Type, data any) error
+
+	attempt int
+	conn    *websocket.Conn
+}
+
+func (c *DevClient) Run(ctx context.Context, addr string, header http.Header) error {
+	if c.Handler == nil {
+		return ErrHandlerNotSet
+	}
+
+	if c.Logger == nil {
+		c.Logger = log.NewNoopLogger()
+	}
+
+	for ctx.Err() == nil {
+		if err := c.dialBackoff(ctx, addr, nil); err != nil {
```

Comment:
Is there ever a context deadline set for the context that is passed into this method?  I can't see any.  In order to ensure that `dialBackoff` has some kind of meaningful timeout, it might be worth it to create a child context with a default deadline.  Or then again, maybe not if it is the intention of this context to never timeout and only to have been cancelled by some other means.

Reply:
There is no deadline; instead, it's managed by an interrupt signal by the main. My idea here was to keep this running forever, so when you restart `gnodev`, it automatically reconnects at some point. To be honest, I'm not really sure of this flow, but since the development is the main purpose of `gnobro`, I really want this enabled by default